mlflow/utils/credentials.py

```diff
@@ -183,5 +191,18 @@ def _databricks_login():
     profile_name = os.environ.get("DATABRICKS_CONFIG_PROFILE", "DEFAULT")
     _overwrite_or_create_databricks_profile(file_name, profile, profile_name)
 
-    if not _check_databricks_auth():
-        raise MlflowException("Failed to sign in Databricks, please retry `mlflow.login()`.")
+    try:
+        # Failed validation will throw an error.
+        _validate_databricks_auth()
+        # Connect to Databricks if the credentials are valid.
+        set_tracking_uri("databricks")
+        return
+    except subprocess.TimeoutExpired:
+        _logger.error(
+            "Timeout while signing in Databricks, max retry reached. This often happens when you "
+            "are using an invalid host. Please check your credentials and retry `mlflow.login`."
+        )
```

Comment:
This error handler looks confusing. We can define a custom execption if necessary.

```
class MaxAttempts:
```

Reply:
curious - is it confusing because I am catching `TimeoutExpired` and throwing another `TimeoutExpired`?